autoload/ale/fixers/prettier.vim

```diff
@@ -30,9 +30,22 @@ endfunction
 function! ale#fixers#prettier#ApplyFixForVersion(buffer, version_output) abort
     let l:executable = ale#fixers#prettier#GetExecutable(a:buffer)
     let l:options = ale#Var(a:buffer, 'javascript_prettier_options')
-
+    let l:filetype = getbufvar(a:buffer, '&filetype')
     let l:version = ale#semver#GetVersion(l:executable, a:version_output)
 
+    " Append the --parser flag depending on the current filetype (unless it's
+    " already set in g:javascript_prettier_options).
+    if match(l:options, '--parser') == -1
```

Comment:
You should probably only pass the `--parser` option if the file has no file extension.